content/blog/kamailio-consul-service-discovery.md

```diff
@@ -0,0 +1,112 @@
+Title: Kamailio service discovery with Consul
+Date: 2019-12-02
+Author: Fabio Tranchitella
+Category: Wazo Platform C4
+Tags: wazo-platform, c4, consul, kubernetes, cloud, kamailio, sip
+Slug: kamailio-consul-service-discovery
+Status: published
+
+# Using Consul for service discovery in Wazo Platform C4
+
+## Introduction
+
+As we started with the deployment of our C4 (Class 4) SBC and routing solution on a Kubernetes Cluster through a Helm Chart, we knew that scaling would be a core feature.
+
+Dynamic scaling brought additional complexity in the configuration of our Kamailio components. Having a cloud-native solution meant that we had to find a way to make SBCs and routers auto-configure themselves without human intervention or static configurations, which will bound the size of our architecture.
+
+We wanted to deploy Wazo on different kinds of platforms, not only containerized with Kubernetes but also with Docker compose or on Virtual machines using Ansible recipes. We had to choose a tool that will help us in service discovery and the flexible configuration of our components. That's where we decided that [HashiCorp Consul](www.consul.io) was the right tool to use.
+
+## What is Consul?
+
+The [official](www.consul.io) definition of Consul is:
+
+```Consul is a service networking solution to connect and secure services across any runtime platform and public or private cloud```
```

Comment:
nitpicking, > citation instead of ```codeblock``` ?

Reply:
@rgarrigue `docker_entrypoint.sh` is the "entrypoint" which launches the executables. Then we just do a `trap exit_script SIGINT SIGTERM` ;)
Take a look at that bash script and you should understand how we handle the unregister.

Reply:
@rgarrigue  I would add that we use auto-deregister if the health check fails above a threshold.